supernode/services/artworkdownload/task.go

```diff
@@ -0,0 +1,371 @@
+package artworkdownload
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/btcsuite/btcutil/base58"
+	"golang.org/x/crypto/sha3"
+
+	"github.com/pastelnetwork/gonode/common/errors"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/common/service/task"
+	"github.com/pastelnetwork/gonode/common/service/task/state"
+	"github.com/pastelnetwork/gonode/pastel"
+	rqnode "github.com/pastelnetwork/gonode/raptorq/node"
+)
+
+const (
+	minRQIDsFileLine = 4
+	rqSymbolsDirName = "symbols"
+)
+
+// Task is the task of registering new artwork.
+type Task struct {
+	task.Task
+	*Service
+
+	RQSymbolsDir string
+}
+
+// Run starts the task
+func (task *Task) Run(ctx context.Context) error {
+	ctx = task.context(ctx)
+	defer task.UpdateStatus(StatusTaskCompleted)
+	log.WithContext(ctx).Debug("Start task")
+	defer log.WithContext(ctx).Debug("End task")
+	defer task.Cancel()
+
+	task.SetStatusNotifyFunc(func(status *state.Status) {
+		log.WithContext(ctx).WithField("status", status.String()).Debugf("States updated")
+	})
+
+	return task.RunAction(ctx)
+}
+
+// Download downloads image and return the image.
+func (task *Task) Download(_ context.Context, txid, timestamp, signature, ttxid string) ([]byte, error) {
+	var err error
+	if err = task.RequiredStatus(StatusTaskStarted); err != nil {
+		return nil, err
+	}
+
+	// Create directory for storing symbols of Artwork
+	// task.RQSymbolsDir = path.Join(task.Service.config.RqFilesDir, task.ID(), rqSymbolsDirName)
+	// err = os.MkdirAll(task.RQSymbolsDir, os.ModePerm)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	var file []byte
+	var artRegTicket pastel.RegTicket
+
+	<-task.NewAction(func(ctx context.Context) error {
```

Comment:
this function is quite long, should divide it into smaller functions?

Reply:
yeah, it's divided. Thanks.